kata-boilerplate/src/Game.js

```diff
@@ -0,0 +1,128 @@
+const AreaParser = require('../src/AreaParser')
+const AreaValidator = require('../src/AreaValidator')
+const CellType = require('../src/CellType');
+
+class Game {
+    /**
+     * @param {AreaParser} areaParser 
+     * @param {AreaValidator} areaValidator 
+     */
+    constructor(areaParser, areaValidator, initArea) {
+        this.areaParser = areaParser;
+        this.areaValidator = areaValidator;
+        this.steps = [];
+        this.init(initArea)
+    }
+
+    init(initArea) {
+        const parsedArea = this.areaParser.parse(initArea);
+        if(!AreaValidator.valid(initArea, parsedArea)) {
+            throw Error('Initial area is invalid!');
+        }
+
+        this.steps.push(parsedArea)
+    }
+
+    evolveByNGenerations(n = 1) {
+        for(let i = n; i <= n; i++) {
+            this.nextGeneration()
+        }
+    }
+
+    nextGeneration() {
+        const previousStep = this.steps[this.steps.length - 1];
+
+        let nextStep = this.deepCopy(previousStep)
+
+        for(let x = 0; x < previousStep.length; x++) {
+            for(let y = 0; y < previousStep[x].length; y++) {
+                nextStep[x][y] = this.decide(this.getCellNeighbors(x, y, previousStep), previousStep[x][y])
+            }
+        }
+        this.steps.push(nextStep);
+
+        return this;
+    }
+
+    deepCopy(object) {
+        return JSON.parse(JSON.stringify(object));
+    }
+
+    /**
+     * @param {array} area 
+     */
+    consoleOutputFormat(area) {
+        let output = '';
+        for(let row of area) {
+            output += row.join('')+'\r\n'
+        }
+        return output.trim();
+    }
+
+    getCellNeighbors(x, y, passedArea) {
+
+        let area = [...passedArea]
+
+
+        const neighbors = []
+        if (isUndefined(area[x-1])) {
```

Comment:
thinking with vectors, not arrays.

Assume you have function `getCell([x, y]): DEAD | LIVE | WTF` you can write this code as:

```js
// assume NEIGHBORS = [[1,0] , [-1, 0], ...]]
return NEIGHBORS.map([dx, dy] => [dx + x, dy + y]).map(getCell);
```